chainer/gradient_check.py

```diff
@@ -345,21 +375,23 @@ def run(self):
         # This must be done before sampling a direction vector, because
         # otherwise the shapes of uninitialized parameters wouldn't be
         # determined.
-        xs_backward, y_backward, y0_data, y_grad = (
+        xs_backward, ys = (
             self._forward_for_backward_gradients())
-        self.y_grad = y_grad
+        # Keep output arrays to save computation in numerical gradients
+        y0_data = tuple([y.array for y in ys])
+
+        # If y_grad is not given, generate the all-1 gradients.
```

Comment:
The current behavior is compatible with `Variable.backward`.  Do you want to change it to be compatible with `chainer.grad`?

Reply:
Are you referring to the `Variable.backward` behavior where it fills grad with 1 only for those with shape `()`? Actually it's not so clear to me why the behavior of `Variable.backward` is like that.
Also, considering the purpose of `gradient_check`, I think there's no reason for uers of it to expect the same behavior.

Reply:
random is a better default

Reply:
That's also fine for me, but isn't that compatibility breaking?
In that case ()-shape grads should be initialized in random for consistency.

Reply:
After discussion, I reverted the behavior change. That can be separated from this PR.